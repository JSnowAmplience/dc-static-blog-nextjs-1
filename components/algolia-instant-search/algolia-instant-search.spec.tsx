/* eslint-env jest */
import React from 'react';

import algoliasearch from 'algoliasearch';

import ShallowRenderer from 'react-test-renderer/shallow';
import AlgoliaInstantSearch from './algolia-instant-search';

jest.mock('algoliasearch', () => jest.fn().mockImplementation(() => ({})));
jest.mock('../algolia-search-result-list/algolia-search-result-list');

describe('AlgoliaInstantSearch', () => {
  test('renders component', async () => {
    const component = ShallowRenderer.createRenderer();
    const props = {
      appId: 'test-app-id',
      searchKey: 'test-search-key',
      indexName: 'test-index-name',
      resultsState: {
        testResultStateData: 'test result state value'
      }
    };
    component.render(<AlgoliaInstantSearch {...props} />);

    expect(algoliasearch).toHaveBeenCalledWith(props.appId, props.searchKey);
    expect(component.getRenderOutput()).toMatchSnapshot();
  });
});
