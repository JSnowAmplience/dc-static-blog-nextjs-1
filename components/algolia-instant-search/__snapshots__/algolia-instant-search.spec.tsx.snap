// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlgoliaInstantSearch renders component 1`] = `
<InstantSearch
  indexName="test-index-name"
  resultsState={
    Object {
      "testResultStateData": "test result state value",
    }
  }
  searchClient={Object {}}
>
  <SearchResultList />
</InstantSearch>
`;
